pkg/sql/sem/tree/datum.go

```diff
@@ -650,6 +650,11 @@ func NewDInt(d DInt) *DInt {
 // ParseDInt parses and returns the *DInt Datum value represented by the provided
 // string, or an error if parsing is unsuccessful.
 func ParseDInt(s string) (*DInt, error) {
+	//strip leading zeroes
+	for len(s) > 1 && s[0] == '0' {
```

Comment:
nit: use `strings.TrimLeft(s, "0"))` 
but can we use `strconv.ParseInt(s, 10, 64)` instead?